src/twisted/trial/__main__.py

```diff
@@ -4,6 +4,15 @@
 if __name__ == "__main__":
     import sys
 
-    from pkg_resources import load_entry_point
+    if sys.version_info >= (3, 8):
+        from importlib.metadata import distribution
+    else:
+        from importlib_metadata import distribution
 
-    sys.exit(load_entry_point("Twisted", "console_scripts", "trial")())
+    sys.exit(
+        next(
+            ep
+            for ep in distribution("twisted").entry_points
+            if (ep.group, ep.name) == ("console_scripts", "trial")
+        ).load()()
+    )
```

Comment:
Part 2: let's not overcomplicate things
```suggestion
    from twisted.scripts.trial import run
    sys.exit(run())
```